### PR TITLE
US19 - Refactoring

### PR DESCRIPTION
**Pull request by JudithRe**
As per [US19](https://github.com/JudithRe/capstone-flashcard-dictionary/issues/45)

Cleaned up.

### CODE REVIEW

File: pages/study/index.js [hunk 1]
```diff
@@ -33,46 +42,70 @@ export default function StudyPage({ wordList, databaseMutate }) {
...
+          return isDue(entry.study.lastReview, entry.study.stage);
+        });
+
+        setStudyList(entriesDue);
+
+        return entriesDue;
+      }
+    }
+    generateStudyMode();
+  }, [wordList]);
```
Comment (F-Kirchhoff) [inline pages/study/index.js]: why do you need a useEffect here? Coudn't you just filter the wordList directly?

File: pages/study/index.js [hunk 2]
```diff
@@ -33,46 +42,70 @@ export default function StudyPage({ wordList, databaseMutate }) {
...
-            <StyledStartButton
-              onClick={() =>
-                generateStudyMode({
-                  wordList,
-                  setStudyList,
-                  setIsStudyMode,
-                })
-              }
-            >
+            <p style={{ textAlign: "center" }}>
```
Comment (F-Kirchhoff) [inline pages/study/index.js]: please don't use inline styling

File: utils/studyFunctions.js [hunk 3]
```diff
@@ -144,14 +133,18 @@ async function handleUpdateEntryInStudyMode({
...
-  if ((wasWrongAnswer && streak > 0) || (!wasWrongAnswer && streak < 0)) {
+  if (wasWrongAnswer && streak > 0) {
     return 0;
   }
 
   if (wasWrongAnswer && streak <= 0) {
     return streak - 1;
   }
 
+  if (!wasWrongAnswer && streak < 0) {
```
Comment (F-Kirchhoff) [inline utils/studyFunctions.js]: Shouldn't it be streak <= 0?

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Judith,
that was a very long PR, please divide your refactoring in bigger chunks next time. I added some suggestions what you need to change, but most of it looks fine.

File: components/FlashcardBack/index.js [hunk 4]
```diff
@@ -25,11 +28,7 @@ function FlashcardBack({ entry, studyList, setIsFront, databaseMutate }) {
             handleWrongAnswer({ studyList, setIsFront, entry, databaseMutate })
           }
         >
-          <span
-            className="inherit-background-color"
-            role="img"
-            aria-label="did not know"
-          >
+          <span role="img" aria-label="did not know">
```
Comment (Roland-Hufnagel) [inline components/FlashcardBack/index.js]: Then the screen-reader reads: "Did not know"??

File: components/Navigation/index.js [hunk 5]
```diff
@@ -18,57 +24,59 @@ export default function Navigation() {
   return (
     <nav>
       <StyledNavList>
-        <StyledNavigationItem>
+        <StyledNavigationItem
+          className="navTransition"
```
Comment (Roland-Hufnagel) [inline components/Navigation/index.js]: Is there a reason why you use ``className`` in a styled-component?

**Review by Roland-Hufnagel — COMMENTED** [review]
Your app looks very good! I'm just wondering why you often use className=...  in styled-components.

File: pages/study/index.js (hunk 1 above)
Comment (JudithRe) [reply pages/study/index.js]: I had it without a useEffect before. But it gave me an infinity loop.

File: components/Navigation/index.js (hunk 5 above)
Comment (JudithRe) [reply components/Navigation/index.js]: No, I removed them :)

File: components/FlashcardBack/index.js (hunk 4 above)
Comment (JudithRe) [reply components/FlashcardBack/index.js]: Yeah, probably not too intuitive. I changed it

File: pages/study/index.js (hunk 1 above)
Comment (F-Kirchhoff) [reply pages/study/index.js]: Ah but maybe you don't even need a State for your filtered list. just use the constant "entriesDue" directly

Comment (JudithRe) [reply pages/study/index.js]: It throws me an infinity loop again :(

**Pull request merged**